nucliadb/nucliadb/search/predict.py

```diff
@@ -229,9 +258,24 @@ def get_predict_url(self, endpoint: str) -> str:
         else:
             return f"{self.cluster_url}{PRIVATE_PREDICT}{endpoint}"
 
-    def get_predict_headers(self, kbid: str) -> dict[str, str]:
+    async def get_predict_headers(self, kbid: str) -> dict[str, str]:
         if self.onprem:
-            return {"X-STF-NUAKEY": f"Bearer {self.nuclia_service_account}"}
+            headers = {"X-STF-NUAKEY": f"Bearer {self.nuclia_service_account}"}
+            config = await self.get_configuration(kbid)
+            if self.local_predict_headers is not None:
+                headers.update(self.local_predict_headers)
+            if config is not None:
```

Comment:
they all follow the same naming, you can do a loop with something like: (untested)

```
SFT_HEADERS = ("ANONYMIZATION-MODEL", "SEMANTIC-MODEL", ...)

for header in SFT_HEADERS:
    value = getattr(config, header.lower().replace('-', '_'), None)
    if value:
        headers[f"X-SFT-{header}"]  = value
```

Reply:
We will do in another iteration